web/app.js

```diff
@@ -1117,16 +1117,25 @@ let PDFViewerApplication = {
                   (AppOptions.get('enableWebGL') ? ' [WebGL]' : '') + ')');
 
       let pdfTitle;
-      if (metadata && metadata.has('dc:title')) {
-        let title = metadata.get('dc:title');
-        // Ghostscript sometimes return 'Untitled', sets the title to 'Untitled'
-        if (title !== 'Untitled') {
-          pdfTitle = title;
-        }
-      }
 
-      if (!pdfTitle && info && info['Title']) {
-        pdfTitle = info['Title'];
+      const infoTitle = info && info['Title'];
+      if (infoTitle) {
+        pdfTitle = infoTitle;
+      }
+      const metadataTitle = metadata && metadata.get('dc:title');
+      if (metadataTitle) {
+        // Ghostscript can produce invalid 'dc:title' Metadata entries:
+        //  - The title may be "Untitled" (fixes bug 1031612).
+        //  - The title may contain incorrectly encoded characters, which thus
+        //    looks broken, hence we ignore the Metadata entry when it contains
+        //    characters from the Specials Unicode block (fixes bug 1605526).
+        //
+        //    (To avoid displaying no viewer title at all, always fallback to
+        //     the Metadata entry regardless when no valid Info entry exists.)
```

Comment:
I could easily be convinced to remove this special-case, which was *only* added in an attempt to not change the current behaviour too much!

Reply:
Having thought about this some more I've decided to just skip the special handing for the `!pdfTitle` case, since the viewer title really seems like a best effort solution anyway with no guarantees that it'll be perfect.